eclipse/src/saros/editor/EditorActivityDelayer.java

```diff
@@ -0,0 +1,89 @@
+package saros.editor;
+
+import java.util.Objects;
+import org.apache.log4j.Logger;
+import org.eclipse.swt.widgets.Display;
+import saros.activities.EditorActivity;
+import saros.activities.TextEditActivity;
+import saros.activities.TextSelectionActivity;
+import saros.activities.ViewportActivity;
+import saros.session.AbstractActivityProducer;
+import saros.session.ISarosSession;
+import saros.ui.util.SWTUtils;
+import saros.util.StackTrace;
+
+/**
+ * This class is responsible for delaying and discarding certain activities that are normally
+ * generated frequently.
+ */
+// TODO Viewport activities are also fire to frequently, worst case is selecting text with mouse
+// while scrolling up or down
+
+final class EditorActivityDelayer extends AbstractActivityProducer {
+
+  private static final int DELAY = 100; // ms
+
+  private static final Logger log = Logger.getLogger(EditorActivityDelayer.class);
+
+  private final Display display;
+
+  private TextSelectionActivity currentSelectionActivity = null;
+
+  EditorActivityDelayer() {
+    display = SWTUtils.getDisplay();
```

Comment:
This is the reason the builds fail. It is trying to load the display without any workbench.

Could you request the display on the fly in the method calls instead to avoid the issues with the context tests? It is *very* annoying to mock the SWT stuff.

I got it to work for a single test class, but when executing the test suites, it complains that the SWT library is already loaded by another classloader (as it wants to execute `SarosEclipseContextTest` and `SarosEclipseContextFactoryTest`). And I have no idea how to solve this.

I tried to get PowerMock to suppress or mock the CTOR call to `EditorManager` or `EditorActivityDelayer` without any success.

Reply:
The fucking tests should use the IEditorManager and not with the EditorManager. Do not know who was so smart to actually try to mockup the EditorManager.